src/policy_evaluator.rs

```diff
@@ -186,32 +232,77 @@ impl PolicyEvaluator {
                 );
             }
         };
-        match self.wapc_host.call("validate", validate_str.as_bytes()) {
-            Ok(res) => {
-                let pol_val_resp: Result<PolicyValidationResponse> = serde_json::from_slice(&res)
-                    .map_err(|e| anyhow!("cannot deserialize policy validation response: {:?}", e));
-                pol_val_resp
-                    .and_then(|pol_val_resp| {
-                        ValidationResponse::from_policy_validation_response(
-                            uid.to_string(),
-                            &req_obj,
-                            &pol_val_resp,
-                        )
-                    })
-                    .unwrap_or_else(|e| {
+        match &mut self.runtime {
```

Comment:
I don't like to have this `switch` statement that goes over all the possible runtimes, to then execute appropriate code.

Can't we define a trait, let's say `validate`, and then have the waPC and the Burrego runtimes implement that? This would dry this code quite a bit

Reply:
Makes a lot of sense. Going to add it.

Reply:
Looking at the code I'm dubious now. Is the same type: the enum Runtime. Thus, adding a trait here is not going to help: is not a case of different types with same behavior, but a single type with different behaviors based on the enum value.

Reply:
Resolving. https://github.com/kubewarden/policy-evaluator/issues/24 is filed and we can continue over there.